src/core/friendly_errors/sketch_reader.js

```diff
@@ -0,0 +1,272 @@
+    const p5Constructors = {};
+    for (let key of Object.keys(p5)) {
+      // Get a list of all constructors in p5. They are functions whose names
+      // start with a capital letter
+      if (typeof p5[key] === 'function' && key[0] !== key[0].toLowerCase()) {
+        p5Constructors[key] = p5[key];
+      }
+    }
+    for (let i = 0; i < funcs.length; i++) {
+      //for every function name obtained check if it matches any p5.js function name
+      const keyArr = Object.keys(p5Constructors);
+      let j = 0;
+      for (; j < keyArr.length; j++) {
+        if (p5Constructors[keyArr[j]].prototype[funcs[i]] !== undefined) {
+          //if a p5.js function is used ie it is in the funcs array
+          p5._friendlyError(
+            translator('fes.sketchReaderErrors.reservedFunc', {
+              symbol: funcs[i]
+            })
+          );
+          break;
+        }
+      }
+      if (j < keyArr.length) break;
+    }
+  };
+
+  /**
+   * Takes an array in which each element is a line of code
+   * containing a variable definition(Eg: arr=['let x = 100', 'const y = 200'])
+   * and extracts the variables defined.
+   *
+   * @method extractVariables
+   * @private
+   * @param {Array} arr array of lines of code
+   */
+  const extractVariables = arr => {
+    //extract variable names from the user's code
+    let matches = [];
+    arr.forEach(ele => {
+      //extract a, b, c from let/const a=10, b=20, c;
+      if (ele.includes(',')) {
+        matches.push(
+          ...ele.split(',').flatMap(s => {
+            let match;
+            if (s.includes('=')) {
+              match = s.match(/(\w+)\s*(?==)/i);
+              return match[1];
+            } else if (!s.match(new RegExp('[[]{}]'))) {
+              let m = s.match(/(?:(?:let|const|var)\s+)?([\w$]+)/);
+              if (m !== null)
+                return s.match(/(?:(?:let|const|var)\s+)?([\w$,]+)/)[1];
+            } else return [];
+          })
+        );
+      } else {
+        //extract a from let/const a=10;
+        const reg = /(?:(?:let|const)\s+)([\w$]+)/g;
+        const found = ele.matchAll(reg);
+        for (const match of found) {
+          matches.push(match[1]);
+        }
+      }
+    });
+    //check if the obtained variables are a part of p5.js or not
+    checkForConsts(matches); //check if the obtained variables are a p5.js constants
+  };
+
+  /**
+   * Takes an array in which each element is a line of code
+   * containing a function definition(arr=['let x = () => {...}'])
+   * and extracts the functions defined.
+   *
+   * @method extractFuncVariables
+   * @private
+   * @param {Array} arr array of lines of code
+   */
+  const extractFuncVariables = arr => {
+    let matches = [];
+    //extract function names
+    const reg = /(?:(?:let|const)\s+)([\w$]+)/;
+    arr.forEach(ele => {
+      matches.push(ele.match(reg)[1]);
+    });
+    //matches array contains the names of the functions
+    checkForFuncs(matches);
+  };
+
```

Comment:
Looks good! Does `ele.match(reg)` always have a `[1]` element?

Reply:
yes, it is because we are using capture groups here and they return an array with 2 elements in them and the 2nd element is the one that we require.